src/restorer/mnemonic-restorer.ts

```diff
@@ -22,36 +22,42 @@ function restorerFromEsplora<R extends MasterPublicKey>(
     gapLimit = 20,
   }) => {
     const restoreFunc = async function(
-      nextAddrFunc: (index: number) => string
-    ): Promise<number> {
+      nextAddrFunc: (index?: number) => string | undefined
```

Comment:
I do not understand how these changes work? it seems that there is no case where the `nextAddrFunc` is called with `index === undefined`.